backend/src/main/java/com/exadel/carinsurance/service/implementation/AssignmentService.java

```diff
@@ -79,6 +88,39 @@ public ResponseEntity createAssignment( @RequestBody AssignmentRequestEntity req
             );
 
     //  Vehicle condition info creation
+    String namesOfPhotosOfImpact = "";
+
+    try {
+      String directoryPath = String
+          .format(
+              "photosofimpact/userId=%s/assignmentId=%s",
+              user.getUserId(), assignmentEntityFromDB.getAssignmentId()
+          );
+      Path directory = Paths.get( directoryPath );
+
+      if ( !Files.exists( directory ) ) {
+        Files.createDirectories( directory );
+      }
+
+      for ( MultipartFile photo : photosOfImpact ) {
+        String filename = photo.getOriginalFilename();
+
+        if ( "".equals( namesOfPhotosOfImpact ) ) {
+          namesOfPhotosOfImpact = filename;
+        } else {
+          namesOfPhotosOfImpact = String.format( "%s;%s", namesOfPhotosOfImpact, filename );
+        }
+
+        Path filePath = directory.resolve( filename );
+        photo.transferTo( filePath );
+      }

```

Comment:
better to extrat it to photos speciffic service

Reply:
Done